scripts/deploy-on-kind.sh

```diff
@@ -75,13 +111,21 @@ deploy_cf_k8s_api() {
       make docker-build-api
     fi
     kind load docker-image --name "$cluster" "$IMG_API"
-    make deploy-api-kind-auth
+    if [ -n "$use_local_registry" ]; then
+      make deploy-api-kind-local
+    else
+      make deploy-api-kind-auth
```

Comment:
I'm wondering if we should have the `authEnabled: true` bit from `make deploy-api-kind-auth` in the API config that `make deploy-api-kind-local` uses. That's the feature flag we're using for whether or not endpoints should use the user's provided auth token when interacting with the K8s API.

https://github.com/cloudfoundry/cf-k8s-controllers/blob/main/api/config/overlays/kind-auth-enabled/apiconfig/cf_k8s_api_config.yaml#L11

Reply:
We just tried to set `authEnabled: true` for the local kind cluster, and we ran into issues creating orgs:

```
$ cf create-org o
Creating org o as kubernetes-admin...
OK

TIP: Use 'cf target -o "o"' to target new org

$ cf create-space -o o s
Organization 'o' not found.
FAILED
```

Any idea what's going on? The SubnamespaceAnchor was created for the org, but for whatever reason we can't see it as our user.

Reply:
We're thinking this is a bug in the current implementation of org creation and listing. The code appears to create the org as a privileged client, but the fetch it as the actual user. Still, it seems like it should work since our user is an admin, but it isn't 🤷

Reply:
> Any idea what's going on?

No, but I remember having this issue back in time but was in a hurry and forgot to create an issue. I worked it around via 

```
create create-org o
cf target -o o
cf create-space s
cf target -o o -s s
```

No idea why that worked though...

Reply:
@danail-branekov We found that we consistently hit the issue with the option enabled, and didn't see it without, even when we used the four steps above.

Maybe there is some kind of caching of org lists in the client? We noticed that the org is created by the privileged user and then retrieved by the not-privileged user in the case where the option is enabled. I also saw that an old org previously created *did* show up in `cf orgs` but a new one did not.

Reply:
@danail-branekov I filed #392 to describe the `cf orgs` bug. Oddly, it appears to affect any org without spaces in it, but once there is a space in the org, it becomes visible/usable.